explorer/interpreter/resolve_names.cpp

```diff
@@ -239,6 +239,14 @@ static auto ResolveNames(Expression& expression,
       auto& where = cast<WhereExpression>(expression);
       CARBON_RETURN_IF_ERROR(
           ResolveNames(where.self_binding().type(), enclosing_scope));
+      // If we're already in a `.Self` context, remember it so that we can
+      // reuse its value for the inner `.Self`.
+      if (auto enclosing_dot_self =
```

Comment:
Note about the aforementioned comment, this could avoid calling set_enclosing_dot_self by first checking if a value was already set and avoiding Resolve if so.

Reply:
Yeah. I'm leaving as-is for now to match the other setters called by name resolution (eg, `IdentifierExpression::set_value_node`) as the general approach here seems to be to do things twice and check that you get the same results both times.